### PR TITLE
Remove some usage of `DUMMY_HIR_ID`

### PR DESCRIPTION
**Pull request by marmeladema**
Use `ObligationClause::dummy()` when appropriate or replace `hir::DUMMY_HIR_ID`by `hir::CRATE_HIR_ID`, as used in `ObligationClause::dummy()`.

### CODE REVIEW

**Comment by marmeladema** [comment]
cc @eddyb 
Its a (small) attempt to reduce the usage of `DUMMY_HIR_ID`

File: src/librustc_trait_selection/traits/mod.rs [hunk 1]
```diff
@@ -136,7 +136,7 @@ pub fn type_known_to_meet_bound_modulo_regions<'a, 'tcx>(
     let trait_ref = ty::TraitRef { def_id, substs: infcx.tcx.mk_substs_trait(ty, &[]) };
     let obligation = Obligation {
         param_env,
-        cause: ObligationCause::misc(span, hir::DUMMY_HIR_ID),
+        cause: ObligationCause::misc(span, hir::CRATE_HIR_ID),
```
Comment (eddyb) [inline src/librustc_trait_selection/traits/mod.rs:139]: cc @nikomatsakis Is this a good way to encode this?

Comment (eddyb) [reply src/librustc_trait_selection/traits/mod.rs:139]: Oh this is what `ObligationCause::dummy` does, heh.

**Review by eddyb — APPROVED** [review]
r=me after it passes tests (when not broken, GHA PR CI is probably the thing to watch)

**EDIT**: oh this PR isn't a few minutes old, and it has already passed.

File: src/librustc_trait_selection/traits/mod.rs (hunk 1 above)
Comment (marmeladema) [reply src/librustc_trait_selection/traits/mod.rs:139]: Yes, i just "copied" what was done in `dummy`

File: src/librustc_passes/check_attr.rs [hunk 2]
```diff
@@ -360,7 +359,7 @@ impl CheckAttrVisitor<'tcx> {
         if let hir::StmtKind::Local(ref l) = stmt.kind {
             for attr in l.attrs.iter() {
                 if attr.check_name(sym::inline) {
-                    self.check_inline(DUMMY_HIR_ID, attr, &stmt.span, Target::Statement);
+                    self.check_inline(l.hir_id, attr, &stmt.span, Target::Statement);
```
Comment (eddyb) [inline src/librustc_passes/check_attr.rs:362]: Heh this is a bugfix (`#[allow]` & friends will now work correctly).

**Review by eddyb — APPROVED** [review]

**Pull request merged**